examples/nlp/sentence_embeddings_with_sbert.py

```diff
@@ -0,0 +1,472 @@
+        # Only average the unpadded output.
+        mask = tf.expand_dims(tf.cast(padding_mask, "float16"), axis=-1)
+        count = tf.math.reduce_sum(mask, axis=1)
+        sum = tf.math.reduce_sum(last_state * mask, axis=1)
+        pooled = sum / count
+        return pooled
+
+
+preprocessor = keras_nlp.models.RobertaPreprocessor.from_preset("roberta_base_en")
+backbone = keras_nlp.models.RobertaBackbone.from_preset("roberta_base_en")
+inputs = keras.Input(shape=(1), dtype="string", name="sentence")
+x = preprocessor(inputs)
+h = backbone(x)
+embedding = mean_pooling(name="pooling_layer")(h, x["padding_mask"])
+n_embedding = tf.linalg.normalize(embedding, axis=1)[0]
+roberta_encoder = keras.Model(inputs=inputs, outputs=n_embedding)
+
+roberta_encoder.summary()
+
+"""
+#### Build the siamese network with regression objective function
+
+It's described above that the Siamese network has two or more subnetworks, and for this
+Siamese model, we need two encoders. But we don't have two encoders; we have only one
+encoder, but we will pass the two sentences through it. That way, we can have two paths
+to get the embeddings and also shared weights between the two paths.
+
+After passing the two sentences to the model and getting the normalized embeddings, we
+will multiply the two normalized embeddings to get the cosine similarity between the two
+sentences.
+"""
+
+
+class regression_siamese(keras.Model):
+    def __init__(self, encoder, **kwargs):
+        inputs = keras.Input(shape=(2), dtype="string", name="sentences")
+        sen1, sen2 = tf.split(inputs, num_or_size_splits=2, axis=1, name="split")
+        u = encoder(sen1)
+        v = encoder(sen2)
+        cosine_similarity = tf.matmul(u, tf.transpose(v))
+
+        super().__init__(
+            inputs=inputs,
+            outputs=cosine_similarity,
+            **kwargs,
+        )
+
+        self.encoder = encoder
+
+    def get_encoder(self):
+        return self.encoder
+
+
+"""
+#### Fit the model to the data
+"""
+
+"""
+Let's try this example before training and compare it to the output after training.
+"""
+
+sentences = [
+    "today is a very sunny day",
+    "I am hungry, I will get my meal.",
+    "the dog is eating his food.",
+]
+query = ["the dog is enjoying his meal."]
+
+encoder = roberta_encoder
+
+sentence_embeddings = encoder(tf.constant(sentences))
+query_embedding = encoder(tf.constant(query))
+
+cosine_simalarities = tf.matmul(query_embedding, tf.transpose(sentence_embeddings))
+for i, sim in enumerate(cosine_simalarities[0]):
+    print(f"cosine similarity between sentence {i+1} and the query = {sim} ")
+
+"""
+For the training, we will use `MeanSquaredError()` loss function and `Adam()` optimizer
+with learning rate = 2e-5.
+"""
+
+roberta_siamese = regression_siamese(roberta_encoder)
+roberta_siamese.compile(
+    loss=keras.losses.MeanSquaredError(),
+    optimizer=keras.optimizers.Adam(2e-5),
+    metrics=[keras.metrics.MeanSquaredError()],
```

Comment:
probably pass `jit_compile=True` here, generally speaking that will be a performance increase, and eventually the default here at least for GPU/TPU

Reply:
`jit_compile=True` make `fit()` return an error